unitTests/ui/ResultsPerPageTest.ts

```diff
@@ -48,6 +49,55 @@ export function ResultsPerPageTest() {
       });
     });
 
+    describe('when modifying the query state model', () => {
+      const setQueryStateModelValue = (value: any) => {
+        test.env.queryStateModel.set(QUERY_STATE_ATTRIBUTES.NUMBER_OF_RESULTS, value);
+      };
+
+      const verifyStateForNumberOfResults = (value: any) => {
+        expect(test.cmp.resultsPerPage).toBe(value);
+        expect(test.env.queryStateModel.get(QUERY_STATE_ATTRIBUTES.NUMBER_OF_RESULTS)).toBe(value);
+      };
+
+      beforeEach(() => {
+        const advancedOptions = <IResultsPerPageOptions>{
+          choicesDisplayed: [10, 25, 50, 100],
+          initialChoice: 25
+        };
+
+        const advancedSetup = new Mock.AdvancedComponentSetupOptions(null, advancedOptions, builder => {
+          return builder.withLiveQueryStateModel();
+        });
+
+        test = Mock.advancedComponentSetup<ResultsPerPage>(ResultsPerPage, advancedSetup);
+        Simulate.initialization(test.env);
+      });
+
+      it('should accept setting the value to the original default 10', () => {
```

Comment:
Hmm, I don't see the difference between this test and the one on line 82? Both seem to set the results per page to a valid value?

Reply:
`10` is a special case since it is the "true" ultimate fallback, and the default value in the query state model.

The bug was originally only present if you configured initial-choice to say `25`, then clicked on `10` (which is the default value in the query state model) and the logic in `getInitialChoice()` would fail and the selection would be ignored.